src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/HtmlCSharp/OnAutoInsertHandler.cs

```diff
@@ -109,21 +109,48 @@ public async Task<DocumentOnAutoInsertResponseItem> HandleRequestAsync(DocumentO
                 return null;
             }
 
-            var remappedEdit = await _documentMappingProvider.RemapFormattedTextEditsAsync(
+            var containsSnippet = response.TextEditFormat == InsertTextFormat.Snippet;
+
+            var edit = response.TextEdit;
+            if (containsSnippet && projectionResult.LanguageKind == RazorLanguageKind.CSharp)
```

Comment:
Can we encompass the snippet wrapping/unwrapping into two methods with a comment above them indicating why we need to do it for future onlookers.